src/lib/registration/widgets/addons_selector.rb

```diff
@@ -0,0 +1,268 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "forwardable"
+require "cwm/multi_status_selector"
+
+module Registration
+  module Widgets
+    # The addons multi selector widget, capable to deal with both
+    #
+    #   - the state (enable or disable)
+    #   - the status (selected/registered, not selected, or auto selected)
+    #
+    # @see CWM::MultiStatusSelector
+    class AddonsSelector < CWM::MultiStatusSelector
```

Comment:
I haven't follow the discussion in this PR, but I would say that `AddonsSelector` should be a `CustomWidget` containing a `MultiStatusSelector` internally.

Reply:
In other words, using composition instead of inheritance from `MultiStatusSelector`.

Reply:
Thanks for the review @joseivanlopez

Reading the comment, probably you're right. The problem is that I did it with the inheritance in mind, according with the agreed in https://gist.github.com/joseivanlopez/f520017cb0effe70887a4447562b0339#file-multi_state_selection_box-rb-L73

I'm not against in changing it for improving it, but this task already consumed a lot of time and the team is waiting for it to make the SP2 branching.

So, if it does not hurt too much, I think we should go ahead postponing the improvements for a next iterations.